sdk/internal/recording/recording_test.go

```diff
@@ -368,7 +372,148 @@ func (s *recordingTests) TestRecordRequestsAndFailMatchingForMissingRecording()
+
+	r.UseHTTPS = false
+	require.Equal(r.HostScheme(), "http://localhost:5000")
+
+	require.Equal(GetEnvVariable(s.T(), "Nonexistentevnvar", "somefakevalue"), "somefakevalue")
+	require.Equal(InPlayback(), !InRecord())
+	require.NotEqual(GetEnvVariable(s.T(), "PROXY_CERT", "fake/path/to/proxycert"), "fake/path/to/proxycert")
+}
+
+var packagePath = "sdk/internal/recording"
+
+func (s *recordingTests) TestStartStop() {
+	require := require.New(s.T())
+
+	os.Setenv("AZURE_RECORD_MODE", "record")
+	defer os.Unsetenv("AZURE_RECORD_MODE")
+
+	err := StartRecording(s.T(), packagePath, nil)
+	require.NoError(err)
+
+	client, err := GetHTTPClient()
+	require.NoError(err)
+
+	req, err := http.NewRequest("POST", "https://localhost:5001", nil)
+	require.NoError(err)
+
+	req.Header.Set(UpstreamUriHeader, "https://www.bing.com/")
+	req.Header.Set(ModeHeader, GetRecordMode())
+	req.Header.Set(IdHeader, GetRecordingId())
+
+	resp, err := client.Do(req)
+	require.NoError(err)
+	require.NotNil(resp)
+
+	require.NotNil(GetRecordingId())
+
+	err = StopRecording(s.T(), nil)
+	require.NoError(err)
+
+	// Make sure the file is there
+	jsonFile, err := os.Open("./recordings/TestRecording/TestStartStop.json")
+	require.NoError(err)
+	defer jsonFile.Close()
+}
+
+func (s *recordingTests) TestUriSanitizer() {
+	require := require.New(s.T())
+
+	os.Setenv("AZURE_RECORD_MODE", "record")
+	defer os.Unsetenv("AZURE_RECORD_MODE")
+
+	err := StartRecording(s.T(), packagePath, nil)
+	require.NoError(err)
+
+	err = AddUriSanitizer("replacement", "bing", nil)
+	require.NoError(err)
+
+	client, err := GetHTTPClient()
+	require.NoError(err)
+
+	req, err := http.NewRequest("POST", "https://localhost:5001", nil)
+	require.NoError(err)
+
+	req.Header.Set(UpstreamUriHeader, "https://www.bing.com/")
+	req.Header.Set(ModeHeader, GetRecordMode())
+	req.Header.Set(IdHeader, GetRecordingId())
+
+	resp, err := client.Do(req)
+	require.NoError(err)
+	require.NotNil(resp)
+
+	require.NotNil(GetRecordingId())
+
+	err = StopRecording(s.T(), nil)
+	require.NoError(err)
+
+	// Make sure the file is there
+	jsonFile, err := os.Open("./recordings/TestRecording/TestUriSanitizer.json")
+	require.NoError(err)
+	defer jsonFile.Close()
+
+	var data RecordingFileStruct
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	require.NoError(err)
+	err = json.Unmarshal(byteValue, &data)
+	require.NoError(err)
+
+	require.Equal(data.Entries[0].RequestUri, "https://www.replacement.com/")
+}
+
+func TestProxyCert(t *testing.T) {
+	_, err := getRootCas()
+	require.NoError(t, err)
+
+	tempProxyCert, ok := os.LookupEnv("PROXY_CERT")
+	require.True(t, ok)
+	err = os.Unsetenv("PROXY_CERT")
+	require.NoError(t, err)
+
+	_, err = getRootCas()
+	require.NoError(t, err)
+
+	err = os.Setenv("PROXY_CERT", "not/a/path.crt")
+	require.NoError(t, err)
+	_, err = GetHTTPClient()
+	require.Error(t, err)
+
+	os.Setenv("PROXY_CERT", tempProxyCert)
+}
+
+func TestStopRecordingNoStart(t *testing.T) {
+	require := require.New(t)
```

Comment:
That's neat!

Reply:
Saves a bit of code, sometimes I use it and sometimes I don't. No real pattern and I can't figure out if I like the option or not.